lib/src/over_react_redux/hooks/use_selector.dart

```diff
@@ -0,0 +1,183 @@
+}
+
+@JS('ReactRedux.useSelector')
+external ReactInteropValue _jsUseSelector(_JsSelectorFn selector, [_JsReduxStateEqualityFn equalityFn]);
+
+// ----------------------------------------------------
+//  createSelectorHook
+// ----------------------------------------------------
+
+/// Returns a function that can be named and used as a replacement for [useSelector] when you want to
+/// use custom [context].
+///
+/// This is useful if you're building a complex reusable component, and you don't want your `Store` to collide
+/// with any Redux `Store` your consumers' applications might use.
+///
+/// See the [react-redux JS documentation](https://react-redux.js.org/api/hooks#custom-context) for more details.
+///
+/// ### Custom Context Example
+///
+/// ```dart
+/// import 'package:over_react/over_react.dart';
+/// import 'package:over_react/over_react_redux.dart';
+/// import 'package:redux/redux.dart';
+///
+/// // ------------------------------------
+/// //  1. Declare the custom context
+/// // ------------------------------------
+/// final MyContext = createContext<MyState>();
+/// final useMySelector = createSelectorHook(MyContext);
+/// final myStore = Store(myReducer);
+///
+/// // ------------------------------------
+/// //  2. (Optional) Create a custom
+/// //  provider component that uses the
+/// //  context defined above.
+/// // ------------------------------------
+/// mixin MyProviderProps = UiProps with ReduxProviderProps;
+///
+/// UiFactory<MyProviderProps> MyProvider = uiFunction(
+///   (props) {
+///     return (ReduxProvider()
+///       ..addUnconsumedProps(props, const [])
+///       ..context = MyContext
+///       ..store = myStore
+///     )(
+///       props.children,
+///     );
+///   },
+///   $MyProviderConfig, // ignore: undefined_identifier
+/// );
+///
+/// // ------------------------------------
+/// //  3. Create a function component that
+/// //  uses `useMySelector`.
+/// // ------------------------------------
+/// mixin MyComponentProps on UiProps {}
+///
+/// UiFactory<MyComponentProps> MyComponent = uiFunction(
+///   (props) {
+///     final count = useMySelector<int, MyState>((state) => state.count);
+///
+///     return Dom.div()('The current count is $count');
+///   },
+///   $MyComponentConfig, // ignore: undefined_identifier
+/// );
+///
+/// // ------------------------------------
+/// //  4. Render the function component
+/// //  nested within the custom provider.
+/// // ------------------------------------
+/// main() {
+///   final app = MyProvider()(
+///     MyComponent()(),
+///   );
+///
+///   react_dom.render(app, querySelector('#id_of_mount_node'));
+/// }
+/// ```
+_SelectorFnHook<TReduxState> createSelectorHook<TReduxState>([Context<TReduxState> context]) {
+  final jsHook = _jsCreateSelectorHook(context?.jsThis ?? JsReactRedux.ReactReduxContext);
+
+  return <TSelector>(selector, [equalityFn]) {
```

Comment:
#nit it would be nice if this were a local function, to provide better stack traces if something throws